File: Data Structures & Algorithms/find-minimum-in-rotated-sorted-array/submission-3.ts
class Solution {
    /**
     * @param {number[]} nums
     * @return {number}
     */
    findMin(nums: number[]): number {
        let min: number = nums[0];
        let left: number = 0;
        let right: number = nums.length - 1;

        while (left <= right) {
            if (nums[left] <= nums[right]) {
                min = Math.min(min, nums[left]);
                break;
            }

            const middle = left + Math.floor((right - left) / 2);
            min = Math.min(min, nums[middle]);
            if (nums[middle] >= nums[left]) {
                left = middle + 1;
            } else {
                right = middle - 1;
            }
        }

        return min;
    }
}

File: Data Structures & Algorithms/buy-and-sell-crypto/submission-3.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        if len(prices) < 2:
            return 0

        max_profit: int = 0

        left: int = 0;
        right: int = 1;

        while right < len(prices):
            profit: int = prices[right] - prices[left]

            if profit > 0:
                max_profit = max(max_profit, profit)
            
            if prices[right] < prices[left]:
                left = right
            
            right += 1
        
        return max_profit